lib/streamlit/config.py

```diff
@@ -997,13 +988,14 @@ def _check_conflicts():
         )
 
     # XSRF conflicts
-
     if get_option("server.enableXsrfProtection"):
-        if not get_option("server.enableCORS") or get_option("global.useNode"):
+        if not get_option("server.enableCORS") or get_option("global.developmentMode"):
             LOGGER.warning(
-                "The config option 'server.enableXsrfProtection is not compatible with "
-                "'server.enableCORS'. If 'server.enableXsrfProtection' is on and 'server.enableCORS' "
-                "is off at the same time, we will prioritize 'server.enableXsrfProtection'."
+                "In order to protect against CSRF attacks, we send a cookie with each request. "
+                "To do so, we must specify allowable origins which places a restriction on "
+                "cross origin resource sharing. We will prioritize 'server.enableXsrfProtection' "
+                "over 'server.enableCORS'. If cross origin resource sharing is required, please "
+                "disable 'server.enableXsrfProtection'."
```

Comment:
This is great. Just a few tweaks:

> Warning: config option server.enableCORS=false is not compatible with server.enableXsrfProtection=true.
> So we turned server.enableCORS to true.
>
> More information:
> In order to protect against CSRF attacks, we send a cookie with each request.
> To do so, we must specify allowable origins, which places a restriction on 
> cross-origin resource sharing.
>
> If cross origin resource sharing is required, please disable server.enableXsrfProtection.

Reply:
(Please make sure the printout in the terminal includes newlines, like above.)

Reply:
Minor nit: the Warning is worded a bit strangely. I would change it to:

> Warning: the config option 'server.enableCORS=false' is not compatible with 'server.enableXsrfProtection=true'.
As a result, 'server.enableCORS' is being overridden to 'true'.